feat(dnsservices): Secondary zones resource (#3904)
IBM-Cloud/terraform-provider-ibm: website/docs/r/dns_custom_resolver_secondary_zone.html.markdown

---
subcategory: "DNS Services"
layout: "ibm"
page_title: "IBM : dns_custom_resolver_secondary_zone"
description: |-
  Manages IBM Private DNS custom resolver secondary zone.
---

# ibm_dns_custom_resolver_secondary_zone

The DNS custom resolver secondary zone resource allows users to request and manage secondary zones for a given custom resolver. By creating and enabling a secondary zone resource for a custom resolver, DNS records for a given DNS zone will be transferred from a user provided primary DNS server (on premise) to a private DNS custom resolver hosted on an IBM Cloud VPC. This framework will improve the availability, speed, and security of DNS queries for a given DNS zone.

## Example usage

```
data "ibm_resource_group" "rg" {
  is_default = true
}

# create a VPC for the subnets
resource "ibm_is_vpc" "test-pdns-cr-vpc" {
  depends_on     = [data.ibm_resource_group.rg]

>   name           = "moises-sdk-testing-cross-account-vpc"

Reviewer: it’s better if we keep some generic name instead of people names.

  vpc             = ibm_is_vpc.test-pdns-cr-vpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
  resource_group  = data.ibm_resource_group.rg.id
}

resource "ibm_is_subnet" "test-pdns-cr-subnet2" {
  name            = "moises-sdk-testing-cross-account-subnet2"
  vpc             = ibm_is_vpc.test-pdns-cr-vpc.id
  zone            = "us-south-2"
  ipv4_cidr_block = "10.240.64.0/24"
  resource_group  = data.ibm_resource_group.rg.id
}

# create a DNS instance
resource "ibm_resource_instance" "test-pdns-cr-instance" {
  name              = "moises-sdk-testing-cross-account-dns1"
  resource_group_id = data.ibm_resource_group.rg.id
  location          = "global"
  service           = "dns-svcs"
  plan              = "standard-dns"
}

# create a custom resolver
resource "ibm_dns_custom_resolver" "test" {
  name        = "msz-test-cr2"
  instance_id = ibm_resource_instance.test-pdns-cr-instance.guid
  description = "new test CR - TF"
  enabled     = true
  locations {
    subnet_crn = ibm_is_subnet.test-pdns-cr-subnet1.crn
    enabled    = true
  }
  locations {
    subnet_crn = ibm_is_subnet.test-pdns-cr-subnet2.crn
    enabled    = true
  }
}

> resource "ibm_dns_zone" "pdns-1-zone" {

Reviewer: This resource block is irrelevant in context with secondary zone

  name        = "moises-zone3.com"
  instance_id = ibm_resource_instance.test-pdns-cr-instance.guid
  description = "testdescription"
  label       = "testlabel"
}

resource "ibm_dns_custom_resolver_secondary_zone" "test" {
  instance_id   = ibm_resource_instance.test-pdns-cr-instance.guid
  resolver_id   = ibm_dns_custom_resolver.test.custom_resolver_id
  zone          = "moises-zone4.com"
  enabled       = true
  transfer_from = ["10.0.0.8"]
}
```

## Argument reference
Review the argument reference that you can specify for your resource. 

> - `instance_id` - (Required, String) The GUID of the private DNS service instance.

Reviewer: ```suggestion
- `instance_id` - (Required, String) The GUID of the DNS Services instance.
```

- `resolver_id` - (Required, String) The GUID of the custom resolver.

> - `zone` - (Required, String) DNS records associated with this DNS zone will be transferred to the custom resolver.

Reviewer: ```suggestion
- `zone` - (Required, String) The name of the zone.
```

> - `enabled`- (Required, Bool) To enable or disable a secondary zone transfer rule. 

Reviewer: ```suggestion
- `enabled`- (Required, Bool) To enable or disable a secondary zone. 
```

> - `transfer_from`- (Required, List of Strings) List of IP addresses. DNS records will be transferred from the primary DNS servers associated with the IP addresses in this list to custom resolvers hosted in an IBM Cloud VPC.

Reviewer 2: ```suggestion
- `transfer_from`- (Required, List of Strings) List of IP addresses. DNS records are transferred from the primary DNS servers associated with the IP addresses in this list to custom resolvers hosted in an IBM Cloud VPC.
```

> - `transfer_from`- (Required, List of Strings) List of IP addresses. DNS records will be transferred from the primary DNS servers associated with the IP addresses in this list to custom resolvers hosted in an IBM Cloud VPC.

Reviewer: ```suggestion
- `transfer_from`- (Required, List of Strings)The addresses of DNS servers where the secondary zone data is transferred from.
```

- `description` - (Optional, String) Descriptive text of the secondary zone.

## Attribute reference
In addition to all argument reference list, you can access the following attribute references after your resource is created. 

- `created_on` - (Timestamp) The time (created On) of the Secondary Zone. 
- `modified_on` - (Timestamp) The time (modified On) of the Secondary Zone.
- `secondary_zone_id` - (String) The unique ID of the private DNS custom resolver secondary zone.

## Import

> The `ibm_dns_custom_resolver_secondary_zone` can be imported by using private DNS instance ID, Custom Resolver ID, and Secondary Zone ID.

Reviewer 2: ```suggestion
The `ibm_dns_custom_resolver_secondary_zone` can be imported by using DNS Services instance ID, Custom Resolver ID, and Secondary Zone ID.
```
